quint/test/static/callgraph.test.ts

```diff
@@ -0,0 +1,159 @@
+import { assert, expect } from 'chai'
+import { describe } from 'mocha'
+
+import { dedent } from '../textUtils'
+import { newIdGenerator } from '../../src/idGenerator'
+import {
+  parsePhase1fromText,
+  parsePhase2sourceResolution,
+  parsePhase3importAndNameResolution,
+} from '../../src/parsing/quintParserFrontend'
+import { SourceLookupPath, fileSourceResolver } from '../../src/parsing/sourceResolver'
+import { LookupTable, QuintDef, QuintExport, QuintImport, QuintInstance, QuintModule } from '../../src'
+import { CallGraphVisitor, mkCallGraphContext } from '../../src/static/callgraph'
+import { walkModule } from '../../src/IRVisitor'
+
+describe('compute call graph', () => {
+  // Parse Quint code without, stopping after name resolution
+  function parse3phases(code: string): [QuintModule[], LookupTable] {
+    const idGen = newIdGenerator()
+    const fakePath: SourceLookupPath = {
+      normalizedPath: 'fake_path',
+      toSourceName: () => 'fake_path',
+    }
+
+    // we are calling parse phases directly instead of `parse`,
+    // as the call graph will be computed at parse phase 4
+    const resolver = fileSourceResolver()
+    const parseResult = parsePhase1fromText(idGen, code, fakePath.normalizedPath)
+      .chain(phase1Data => parsePhase2sourceResolution(idGen, resolver, fakePath, phase1Data))
+      .chain(phase2Data => parsePhase3importAndNameResolution(phase2Data))
+
+    if (parseResult.isLeft()) {
+      const msgs = parseResult.value.map(e => e.explanation).join('; ')
+      assert.fail(`Failed to parse a mock module: ${msgs}`)
+    }
+    const { modules, table } = parseResult.unwrap()
+    return [modules, table]
+  }
+
+  function findDef(module: QuintModule, name: string): QuintDef {
+    const d = module.defs.find(
+      d => (d.kind === 'def' || d.kind === 'const' || d.kind === 'var' || d.kind === 'typedef') && d.name === name
+    )
+    assert(d, `Definition ${name} not found`)
+    return d
+  }
+
+  function findImport(module: QuintModule, pred: (imp: QuintImport) => boolean): QuintDef {
+    const d = module.defs.find(d => d.kind === 'import' && pred(d))
+    assert(d, `Import not found in ${module.name}`)
+    return d
+  }
+
+  function findInstance(module: QuintModule, pred: (imp: QuintInstance) => boolean): QuintDef {
+    const d = module.defs.find(d => d.kind === 'instance' && pred(d))
+    assert(d, `Instance not found in ${module.name}`)
+    return d
+  }
+
+  function findExport(module: QuintModule, pred: (imp: QuintExport) => boolean): QuintDef {
+    const d = module.defs.find(d => d.kind === 'export' && pred(d))
+    assert(d, `Export not found in ${module.name}`)
+    return d
+  }
+
+  it('const, var, and operator definitions', () => {
```

Comment:
I think the BDD idea is to follow `it` with a verb 😉 

```suggestion
  it('computes a call graph of const, var, and operator definitions', () => {
```